numpy/core/src/multiarray/compiled_base.c

```diff
@@ -116,10 +116,10 @@ arr_bincount(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
     }
     else {
         minlength = PyArray_PyIntAsIntp(mlength);
```

Comment:
I suppose we can't do it because backwards compatibility... But the temptation is strong to change the `PyArg_ParseTupleAndKeywords` format string to `"O|On" and get rid of the `mlength` variable and the `None` handling logic altogether...